File: 0028-find-the-index-of-the-first-occurrence-in-a-string/0028-find-the-index-of-the-first-occurrence-in-a-string.js
/**
 * @param {string} haystack
 * @param {string} needle
 * @return {number}
 */
var strStr = function(haystack, needle) {
    for(let i=0;i<haystack.length;i++){
        let s=''
        let j=i
        while(s.length<needle.length){
            s+=haystack[j]
            j++
        }
        if(s===needle){
            return i
        }
    }
    return -1
        
}

File: 0028-find-the-index-of-the-first-occurrence-in-a-string/0028-find-the-index-of-the-first-occurrence-in-a-string.py
class Solution:
    def strStr(self, haystack: str, needle: str) -> int:
        i=0
        while i<len(haystack)-len(needle)+1:
            s=''
            for j in range(i,i+len(needle)):
                s+=haystack[j]
            if s==needle:
                return i
            i+=1
        return -1